react/__tests__/productKit.test.js

```diff
@@ -1,19 +1,46 @@
 import React from 'react'
-import { render } from 'test-utils'
+import { render, fireEvent, act } from '@vtex/test-tools/react'
 import { productMock } from '../__mocks__/productMock'
 import ProductKit from '../index'
 
 describe('<ProductSummary /> component', () => {
-  function renderComponent() {
-    return render(<ProductKit {...productMock} />)
+  function renderComponent(customProps) {
+    return render(<ProductKit {...productMock} {...customProps} />)
   }
 
   it('should be rendered', () => {
     expect(renderComponent()).toBeDefined()
   })
 
   it('should match the snapshot', () => {
-    const { asFragment } = renderComponent()
+    const { asFragment } = renderComponent({
+      allowSwap: false,
+      allowRemoval: false,
+    })
     expect(asFragment()).toMatchSnapshot()
   })
+
+  it('should match the snapshot with removal and swap', () => {
+    const { asFragment } = renderComponent({
+      allowSwap: true,
+      allowRemoval: true,
+    })
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  it('should render the 2 itens', () => {
+    const { getByText } = renderComponent()
+    expect(getByText(/Take all 2 products/)).toBeTruthy()
```

Comment:
This `2` here looks like a magic number if you do not know that the mock that you use has only two products.
To be more readable and reusable, It would be better that you generated a mock with an input that was the number of products, then test with a number choosed here.